### PR TITLE
Readme for Microsoft.Extensions.Http.Diagnostics

### PR DESCRIPTION
**Pull request by xakep139**
null

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/extensions/pull/4674)

### CODE REVIEW

File: src/Libraries/Microsoft.Extensions.Http.Diagnostics/README.md [hunk 1]
```diff
@@ -18,6 +18,99 @@ Or directly in the C# project file:
 </ItemGroup>
 ```
 
+## Usage Example
+
+### HTTP Client Logs Enrichment and Redaction
+
+These components enable enriching and redacting HTTP Client request logs. They remove built-it HTTP Client logging.
+
+In order to use the redaction feature, you need to reference the `Microsoft.Extensions.Compliance.Redaction` package.
```
Comment (xakep139) [inline src/Libraries/Microsoft.Extensions.Http.Diagnostics/README.md:27]: This is non-trivial and we need to improve discoverability of this (reference redaction package in `Http.Diagnostics` package?). I will create a work item to track that

File: src/Libraries/Microsoft.Extensions.Http.Diagnostics/README.md [hunk 2]
```diff
@@ -18,6 +18,99 @@ Or directly in the C# project file:
...
+var builder = Host.CreateApplicationBuilder(args);
+
+// Register IHttpClientFactory:
+builder.Services.AddHttpClient();
+
+// Register redaction services:
+builder.Services.AddRedaction();
+
+// Register latency context services:
+builder.Services.AddLatencyContext();
```
Comment (xakep139) [inline src/Libraries/Microsoft.Extensions.Http.Diagnostics/README.md:104]: This was also non-trivial as DI simply throws an exception if I forget to add this line. We definitely need to improve this.

File: src/Libraries/Microsoft.Extensions.Http.Diagnostics/README.md (hunk 1 above)
Comment (xakep139) [reply src/Libraries/Microsoft.Extensions.Http.Diagnostics/README.md:27]: Created #4675

File: src/Libraries/Microsoft.Extensions.Http.Diagnostics/README.md (hunk 2 above)
Comment (xakep139) [reply src/Libraries/Microsoft.Extensions.Http.Diagnostics/README.md:104]: Tracked under #4675

File: src/Libraries/Microsoft.Extensions.Http.Diagnostics/README.md (hunk 1 above)
Comment (joperezr) [reply src/Libraries/Microsoft.Extensions.Http.Diagnostics/README.md:27]: Yeah I think we should reference the Redaction package automatically. Even when it is not used directly, the idea would be for folks referencing this package to be able to look at your sample code from the README and use it, but if we don't include the package reference they would get an error in the call to AddRedaction.

Comment (joperezr) [reply src/Libraries/Microsoft.Extensions.Http.Diagnostics/README.md:27]: That said, I agree with you we can fix this in a separate change.

**Review by joperezr — APPROVED** [review]
Left a very small comment but this looks great otherwise. Thanks so much for taking care of this @xakep139

File: src/Libraries/Microsoft.Extensions.Http.Diagnostics/README.md [hunk 3]
```diff
@@ -1,6 +1,6 @@
 # Microsoft.Extensions.Http.Diagnostics
 
-Telemetry support for HTTP Client.
+Telemetry support for HTTP Client that allows tracking latency and enriching and redacting log output.
```
Comment (Tratcher) [inline src/Libraries/Microsoft.Extensions.Http.Diagnostics/README.md:3]:
```suggestion
Telemetry support for `HttpClient` that allows tracking latency and enriching and redacting log output.
```

File: src/Libraries/Microsoft.Extensions.Http.Diagnostics/README.md [hunk 4]
```diff
@@ -18,6 +18,99 @@ Or directly in the C# project file:
 </ItemGroup>
 ```
 
+## Usage Example
+
+### HTTP Client Logs Enrichment and Redaction
+
+These components enable enriching and redacting HTTP Client request logs. They remove built-it HTTP Client logging.
```
Comment (Tratcher) [inline src/Libraries/Microsoft.Extensions.Http.Diagnostics/README.md:25]:
```suggestion
These components enable enriching and redacting `HttpClient` request logs. They remove built-it HTTP Client logging.
```

Comment (Tratcher) [reply src/Libraries/Microsoft.Extensions.Http.Diagnostics/README.md:25]: Only HttpClient instances created with the IHttpClientFactory benefit from these features, correct?

**Review by Tratcher — APPROVED** [review]

**Review by geeknoid — APPROVED** [review]

File: src/Libraries/Microsoft.Extensions.Http.Diagnostics/README.md (hunk 4 above)
Comment (xakep139) [reply src/Libraries/Microsoft.Extensions.Http.Diagnostics/README.md:25]: Unfortunately, yes.

File: src/Libraries/Microsoft.Extensions.Http.Diagnostics/README.md [hunk 5]
```diff
@@ -18,6 +18,99 @@ Or directly in the C# project file:
...
+For example:
+
+```csharp
+var builder = Host.CreateApplicationBuilder(args);
+
+// Register IHttpClientFactory:
+builder.Services.AddHttpClient();
+
+// Register redaction services:
+builder.Services.AddRedaction();
```
[the remark this answers is not in the document]
Comment (xakep139) [reply src/Libraries/Microsoft.Extensions.Http.Diagnostics/README.md:47]: I tempted to include that as well, but wasn't sure about it. I will add the line, thanks!

File: src/Libraries/Microsoft.Extensions.Http.Diagnostics/README.md (hunk 3 above)
Comment (xakep139) [reply src/Libraries/Microsoft.Extensions.Http.Diagnostics/README.md]: Applied, thanks!

File: src/Libraries/Microsoft.Extensions.Http.Diagnostics/README.md (hunk 4 above)
Comment (xakep139) [reply src/Libraries/Microsoft.Extensions.Http.Diagnostics/README.md]: Applied the suggestion, thanks!

File: src/Libraries/Microsoft.Extensions.Http.Diagnostics/README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (xakep139) [reply src/Libraries/Microsoft.Extensions.Http.Diagnostics/README.md:47]: Done, thanks!

**Comment by xakep139** [comment]
@joperezr do we still merge PRs into `release/8.0` branch?

**Review by dariusclay — APPROVED** [review]

**Pull request merged**